File: pat-to-ascii/pat-to-ascii.py
# pat-to-ascii.py
# select a pattern in Golly, run this script,
#   and a plaintext ASCII format version of
#   the pattern will be copied to the clipboard
#
# For a variant producing Life Lexicon format, see also
#   http://conwaylife.com/forums/viewtopic.php?p=64427#p64427

import golly as g
r = g.getselrect()
s=""
for y in range(r[3]):
  for x in range(r[2]):
    s+="*" if g.getcell(x+r[0],y+r[1]) == 1 else "."
  s+="\n"
g.setclipstr(s)

File: pat-to-ascii/pat-to-ascii.lua
-- pat-to-ascii.lua
-- select a pattern in Golly, run this script,
--   and a plaintext ASCII format version of
--   the pattern will be copied to the clipboard
--
-- For a variant producing Life Lexicon format, see also
--   http://conwaylife.com/forums/viewtopic.php?p=64427#p64427

local g = golly()
r = g.getselrect()
s=""
for y = 0, r[4] do
  for x = 0, r[3] do
    if g.getcell(x+r[1], y + r[2]) == 1 then
      s = s.."*"
    else
      s = s.."."
    end
  end
  s = s.."\n"
end

g.setclipstr(s)
